### ISSUE TITLE
Add a numpy.decimal type?

### ISSUE THREAD

**Issue by garrypolley** [numpy/numpy#9789]
Would it be okay to add a number.decimal type? It would be a subclass of the Decimal type in Python. 

Initially my implementation would be naive and have no optimizations and the dtype would be object.

The reason for the type is so that `np.round` works a little nicer.  The `np.Decimal` type would have an overridden `round` so that `myNpObject.round()` would work, as well as `np.round(np.Decimal)`.

If this is desired, okay, then I may add a PR with that feature.  And it would related to the current PR I have out here: https://github.com/numpy/numpy/pull/9781

**Comment by eric-wieser** [comment]
Am I right in thinking the sole purpose is to support `np.round`? I think the underlying issue is actually #6248, or at least related. `np.round` should delegate to `obj.__round__` on things it doesn't recognize.

**Comment by eric-wieser** [comment]
I've filed a new issue about supporting `np.round(Decimal(...))` at #9791

**Comment by garrypolley** [comment]
That would work. I think for my near term I'll get working on #9781 and have a custom decimal subclass that has a `.round` on it. And if I have time, I'll try to help with the #9791  and potentially #6248

**Comment by eric-wieser** [comment]
Is `round` needed for #9781?

**Comment by garrypolley** [comment]
It is not needed directly. In our application code I'm rounding the values that come out so that I do not end up over/under by a penny. 

Since I was calling `.round` on the results previously, when I switched it to a `Decimal` it broke, because `Decimal` does not have a `round` method. But, if all the `np.array` types worked with the built-in `round` then I would also be covered.

That's the core of why I was contemplating making a custom `Decimal` type. Some numbers we are getting may be `floats` while some others may be `Decimal`, which is why I took the journey into updating the other financial functions to be able to use `Decimal`. 

Thanks for the quick replies and responsiveness. I appreciate it.
